rpcs3/rpcs3qt/update_manager.cpp

```diff
@@ -0,0 +1,547 @@
+			{
+				m_progress_dialog->close();
+				return true;
+			}
+
+			hash_found = false;
+		}
+		else
+		{
+			return false;
+		}
+	}
+
+	const auto& latest = json_data["latest_build"];
+	if (!latest.isObject())
+	{
+		LOG_ERROR(GENERAL, "[Auto-updater] Json doesn't contain latest_build section");
+		return false;
+	}
+
+	QString os;
+
+#ifdef _WIN32
+	os = "windows";
+#elif defined(__linux__)
+	os = "linux";
+#else
+	LOG_ERROR(GENERAL, "[Auto-updater] Your os isn't currently supported by the auto-updater");
+	return false;
+#endif
+
+	// Check that every bit of info we need is there
+	if (!latest[os].isObject() || !latest[os]["download"].isString() || !latest[os]["size"].isDouble() || !latest[os]["checksum"].isString() || !latest["version"].isString() ||
+	    !latest["datetime"].isString() ||
+	    (hash_found && (!json_data["current_build"].isObject() || !json_data["current_build"]["version"].isString() || !json_data["current_build"]["datetime"].isString())))
+	{
+		LOG_ERROR(GENERAL, "[Auto-updater] Some information seems unavailable");
+		return false;
+	}
+
+	if (hash_found && return_code == 0)
+	{
+		LOG_SUCCESS(GENERAL, "[Auto-updater] RPCS3 is up to date!");
+		m_progress_dialog->close();
+
+		if (!automatic)
+			QMessageBox::information(m_parent, tr("Auto-updater"), tr("Your version is already up to date!"));
+
+		return true;
+	}
+
+	if (hash_found)
+	{
+		// Calculates how old the build is
+		std::string cur_date = json_data["current_build"]["datetime"].toString().toStdString();
+		std::string lts_date = latest["datetime"].toString().toStdString();
+
+		tm cur_tm, lts_tm;
+		QString timediff = "";
+
+		auto time_from_str = [](const std::string& str, const std::string& format, tm* tm) -> bool
+		{
+			std::istringstream input(str);
+			input.imbue(std::locale(setlocale(LC_ALL, nullptr)));
+			input >> std::get_time(tm, format.c_str());
+			if (input.fail())
+				return false;
+
+			return true;
+		};
+
+		if (time_from_str(cur_date, "%Y-%m-%d %H:%M:%S", &cur_tm) && time_from_str(lts_date, "%Y-%m-%d %H:%M:%S", &lts_tm))
+		{
+			time_t cur_time = mktime(&cur_tm);
+			time_t lts_time = mktime(&lts_tm);
+
+			s64 u_timediff = (s64)std::difftime(lts_time, cur_time);
+			timediff       = tr("Your version is %1 day(s), %2 hour(s) and %3 minute(s) old.").arg(u_timediff / (60 * 60 * 24)).arg((u_timediff / (60 * 60)) % 24).arg((u_timediff / 60) % 60);
+		}
+
+		const QString to_show = tr("A new version of RPCS3 is available!\n\nCurrent version: %1\nLatest version: %2\n%3\n\nDo you want to update?")
+		                            .arg(json_data["current_build"]["version"].toString())
+		                            .arg(latest["version"].toString())
+		                            .arg(timediff);
+		if (QMessageBox::question(m_progress_dialog, tr("Update Available"), to_show, QMessageBox::Yes | QMessageBox::No) == QMessageBox::No)
+		{
+			m_progress_dialog->close();
+			return true;
+		}
```

Comment:
You should wrap this in the following two calls for the same reasons as before with the close stuff.
```
m_progress_dialog->hide();
// ...
// this code
// ...
m_progress_dialog->show();
```